packages/gafl-webapp-service/src/pages/renewals/renewal-start-date/__tests__/route.spec.js

```diff
@@ -0,0 +1,181 @@
+  licensee: {
+    noLicenceRequired: {}
+  },
+  renewedEndDate: renewedEndDate
+})
+
+describe('getData', () => {
+  beforeEach(jest.clearAllMocks)
+  it('expiryTimeString is set to return value of displayExpiryDate', async () => {
+    const expiryDate = Symbol('expiryDate')
+    displayExpiryDate.mockReturnValue(expiryDate)
+    const expected = await getData(getMockRequest())
+    expect(expected.expiryTimeString).toBe(expiryDate)
+  })
+
+  it('displayExpiryDate is called with request and permission', async () => {
+    const permission = getSamplePermission()
+    const request = getMockRequest()
+    await getData(request)
+    expect(displayExpiryDate).toHaveBeenCalledWith(request, permission)
+  })
+
+  it('getData returns expected outputs', async () => {
+    const expected = await getData(getMockRequest())
+    expect(expected).toMatchSnapshot()
+  })
+})
+
+describe('setLicenceStartDateAndTime', () => {
+  beforeEach(jest.clearAllMocks)
+
+  it('ageConcessionHelper is called with permission', async () => {
+    const result = {}
+    const permission = { licenceStartDate: '2023-02-11T12:00:00.000Z' }
+    await setLicenceStartDateAndTime(result, getMockRequest(permission))
+    expect(ageConcessionHelper).toHaveBeenCalledWith(permission)
+  })
+
+  it.each([
+    [RENEWAL_START_DATE.uri, 'fails', true],
+    [LICENCE_SUMMARY.uri, 'passes', false]
+  ])('addLanguageCodeToUri is called with request and %s when validation %s', async (uri, desc, error) => {
+    const request = getMockRequest()
+    await setLicenceStartDateAndTime({ error }, request)
+    expect(addLanguageCodeToUri).toHaveBeenCalledWith(request, uri)
+  })
+})
+
+describe('licenceStartTime and licenceToStart values', () => {
+  describe.each`
+    desc                                                                                                                           | endHours | licenceToStartResult | licenceStartDate              | renewedEndDate                | year      | month  | day
+    ${'licenceStartTime = current time and licenceToStart = another date when renewal is current day and licence not yet expired'} | ${13}    | ${'another-date'}    | ${'2023-02-10T12:00:00.000Z'} | ${'2023-02-10T13:00:00.000Z'} | ${'2023'} | ${'2'} | ${'10'}
+    ${'licenceStartTime = 30 minutes and licenceToStart = after payment when renewal is current day and licence is expired'}       | ${null}  | ${'after-payment'}   | ${'2023-02-10T12:00:00.000Z'} | ${'2023-02-10T11:00:00.000Z'} | ${'2023'} | ${'2'} | ${'10'}
+    ${'licenceStartTime = midnight and licenceToStart = another date when renewal is different day so renewing in advance'}        | ${0}     | ${'another-date'}    | ${'2023-02-11T12:00:00.000Z'} | ${'2023-02-10T12:00:00.000Z'} | ${'2023'} | ${'2'} | ${'11'}
+  `('returns values of: $desc', ({ endHours, licenceToStartResult, licenceStartDate, renewedEndDate, year, month, day }) => {
```

Comment:
One last thing - I think your $desc is a bit too complicated. if you split it into `timeDesc` and `licenceStartDesc`, then instead of 'returns values of: $desc', you can do 'returns values of: licenceStartTime = $timeDesc and licenceToStart = $licenceStartDesc'